pkg/vulnsrc/openeuler/openeuler.go

```diff
@@ -0,0 +1,246 @@
+	var cvrfs []Cvrf
+	rootDir := filepath.Join(dir, "vuln-list", eulerDir)
+	err := utils.FileWalk(rootDir, func(r io.Reader, path string) error {
+		var cvrf Cvrf
+		if err := json.NewDecoder(r).Decode(&cvrf); err != nil {
+			return xerrors.Errorf("failed to decode openEuler CVRF JSON: %w %+v", err, cvrf)
+		}
+		cvrfs = append(cvrfs, cvrf)
+		return nil
+	})
+	if err != nil {
+		return xerrors.Errorf("error in openEuler CVRF walk: %w", err)
+	}
+
+	if err = vs.save(cvrfs); err != nil {
+		return xerrors.Errorf("error in openEuler CVRF save: %w", err)
+	}
+
+	return nil
+}
+
+func (vs VulnSrc) save(cvrfs []Cvrf) error {
+	err := vs.dbc.BatchUpdate(func(tx *bolt.Tx) error {
+		return vs.commit(tx, cvrfs)
+	})
+	if err != nil {
+		return xerrors.Errorf("error in batch update: %w", err)
+	}
+	return nil
+}
+
+func (vs VulnSrc) commit(tx *bolt.Tx, cvrfs []Cvrf) error {
+	for _, cvrf := range cvrfs {
+		affectedPkgs := getAffectedPackages(cvrf.ProductTree)
+		if len(affectedPkgs) == 0 {
+			continue
+		}
+
+		for _, affectedPkg := range affectedPkgs {
+			advisory := types.Advisory{
+				FixedVersion: affectedPkg.Package.FixedVersion,
+			}
+
+			if err := vs.dbc.PutDataSource(tx, affectedPkg.OSVer, source); err != nil {
+				return xerrors.Errorf("failed to put data source: %w", err)
+			}
+			if err := vs.dbc.PutAdvisoryDetail(tx, cvrf.Tracking.ID, affectedPkg.Package.Name,
+				[]string{affectedPkg.OSVer}, advisory); err != nil {
+				return xerrors.Errorf("unable to save %s CVRF: %w", affectedPkg.OSVer, err)
+			}
+		}
+
+		var references []string
+		for _, ref := range cvrf.References {
+			references = append(references, ref.URL)
+		}
+
+		severity := types.SeverityUnknown
+		for _, cvuln := range cvrf.Vulnerabilities {
+			for _, threat := range cvuln.Threats {
+				sev := severityFromThreat(threat.Severity)
+				if severity < sev {
+					severity = sev
+				}
+			}
+		}
+
+		vuln := types.VulnerabilityDetail{
+			References:  references,
+			Title:       cvrf.Title,
+			Description: getDetail(cvrf.Notes),
+			Severity:    severity,
+		}
+		if err := vs.dbc.PutVulnerabilityDetail(tx, cvrf.Tracking.ID, source.ID, vuln); err != nil {
+			return xerrors.Errorf("failed to save openEuler CVRF vulnerability: %w", err)
+		}
+
+		// for optimization
+		if err := vs.dbc.PutVulnerabilityID(tx, cvrf.Tracking.ID); err != nil {
+			return xerrors.Errorf("failed to save the vulnerability ID: %w", err)
+		}
+	}
+	return nil
+}
+
+func getAffectedPackages(productTree ProductTree) []AffectedPackage {
+	var pkgs []AffectedPackage
+
+	for _, branch := range productTree.Branches {
+		if branch.Type != "Package Arch" || branch.Name != "aarch64" {
+			continue
+		}
+		for _, production := range branch.Productions {
+			osVer := getOSVersion(production.CPE)
+			if osVer == "" {
+				continue
+			}
+
+			productID := production.ProductID
+			if productID == "" {
+				parts := strings.Split(production.Text, ".oe")
+				productID = parts[0]
+			}
```

Comment:
Can you add example for this in comment?

Reply:
Yes, I have added an example in comment. And you can see them from the followings
https://github.com/aquasecurity/vuln-list/blob/24612eef3897977e4671fa4d4488b60da2bc96df/openeuler/2024/openEuler-SA-2024-1001.json#L76
https://github.com/aquasecurity/vuln-list/blob/24612eef3897977e4671fa4d4488b60da2bc96df/openeuler/2024/openEuler-SA-2024-1001.json#L78